### ISSUE TITLE
builder/amazon: not deleting temporary keypair in v0.11.0

### ISSUE THREAD

**Issue by pmcgrath** [mitchellh/packer#4120]
I'm seeing packer not removing the temp key pair at the completion of an AWS AMI build since i upgraded to v0.11.0

My environment info
```
$ packer version
Packer v0.11.0

$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 15.10
Release:	15.10
Codename:	wily
```

After upgrading to v0.11.0 I see the following output at the bottom
```
==> amazon-ebs: Terminating the source AWS instance...
==> amazon-ebs: Cleaning up any extra volumes...
==> amazon-ebs: No volumes to clean up, skipping
==> amazon-ebs: Deleting temporary security group...
Build 'amazon-ebs' finished.
```
No mention of removing the temporary keypair packer created, when I look in the AWS console I can see key pairs with the packer_ prefix being created but not being removed

I switched back to v0.10.2 and rebuild the same AMI and I can see the following output at the bottom
```
==> amazon-ebs: Tagging snapshot: snap-05b462751a8d4dc9f
==> amazon-ebs: Terminating the source AWS instance...
==> amazon-ebs: Cleaning up any extra volumes...
==> amazon-ebs: No volumes to clean up, skipping
==> amazon-ebs: Deleting temporary security group...
==> amazon-ebs: Deleting temporary keypair...
Build 'amazon-ebs' finished.
```
The temporary keypair that packer created and used for this build and no longer exists

Simple packer template to test this is [here](https://gist.github.com/pmcgrath/f66e9576cb543e8ae652bc9e6dfb1603#file-simple-packer-sample)

This is the bottom of the debug log
```
==> amazon-ebs: Terminating the source AWS instance...
2016/11/04 22:04:04 packer: 2016/11/04 22:04:04 Waiting for state to become: terminated
2016/11/04 22:04:04 packer: 2016/11/04 22:04:04 Allowing 300s to complete (change with AWS_TIMEOUT_SECONDS)
2016/11/04 22:04:11 ui: ==> amazon-ebs: Cleaning up any extra volumes...
==> amazon-ebs: Cleaning up any extra volumes...
2016/11/04 22:04:11 ui: ==> amazon-ebs: No volumes to clean up, skipping
==> amazon-ebs: No volumes to clean up, skipping
2016/11/04 22:04:11 ui: ==> amazon-ebs: Deleting temporary security group...
==> amazon-ebs: Deleting temporary security group...
2016/11/04 22:04:11 ui: Build 'amazon-ebs' finished.
2016/11/04 22:04:11 Builds completed. Waiting on interrupt barrier...
2016/11/04 22:04:11 ui: 
==> Builds finished. The artifacts of successful builds are:
2016/11/04 22:04:11 machine readable: amazon-ebs,artifact-count []string{"1"}
Build 'amazon-ebs' finished.
```

**Comment by mwhooker** [comment]
thanks for the report. I closed #4119 since I was unable to reproduce. Let me try to do some more debugging

**Comment by mwhooker** [comment]
@pmcgrath thanks for the report. This was fixed in https://github.com/mitchellh/packer/pull/4080 which is available on master
